comp/trace/config/config.go

```diff
@@ -46,7 +48,9 @@ type cfg struct {
 	warnings *pkgconfig.Warnings
 }
 
-func newConfig(deps dependencies) (Component, error) {
+// NewConfig is the default constructor for the component, it returns
+// a component instance and an error.
+func NewConfig(deps Dependencies) (Component, error) {
```

Comment:
Why did you changed the visibility of the `newConfig` function and the `dependency` struct?

Reply:
Since all agent-apm changes are just related to the visibility change, I'm also curious about this

Reply:
Synced with @truthbk around this. The exported types are needed here: https://github.com/DataDog/datadog-agent/blob/aef16d5a615caf6b2121814ec79897ca45169777/cmd/otel-agent/subcommands/run/command.go#L188-L194

👍